security/src/main/scala/pl/touk/nussknacker/ui/security/oauth2/CachingOAuth2Service.scala

```diff
@@ -60,5 +60,15 @@ class CachingOAuth2Service[
     userInfo.map { case (userInfo, expiration) => (userInfo, Some(expiration)) }
   }
 
+  override def introspectAccessToken(accessToken: String): Future[IntrospectedAccessTokenData] =
+    throw new IllegalAccessError(
```

Comment:
For me, it's sth suspicious here. It's a decorator and it doesn't allow us to call some public methods on it. Why cannot we just delegate calls to the underlying object (the `delegate`)?

Reply:
I did it because someone can think that one of these methods cache access token as well and this class is designed to be used only via `checkAuthorizationAndObtainUserinfo`. But I changed it as you proposed. Let me know if you change your mind after reading this comment.

Reply:
IMO it's OK now. We could think about changing the name of the class to suggest what is cached, but at the moment, the two methods have a `private [oauth2]` scope, so I don't think we need to do it.